grpc4bmi/bmi_client_singularity.py

```diff
@@ -29,58 +27,160 @@ class BmiClientSingularity(BmiClient):
+
+              # Write config to work_dir and interact with client
+
+              # After checking output in work_dir, clean up
+              work_dir.cleanup()
+
+        delay (int): Seconds to wait for Singularity container to startup, before connecting to it
+
+            Increase when container takes a long time to startup.
+
+        timeout (int): Seconds to wait for gRPC client to connect to server.
+
+            By default will try forever to connect to gRPC server inside container.
+            Set to low number to escape endless wait.
+
+    **Example 1: Config file already inside image**
+
+    MARRMoT has an `example config file <https://github.com/wknoben/MARRMoT/blob/master/BMI/Config/BMI_testcase_m01_BuffaloRiver_TN_USA.mat>`_ inside its Docker image.
+
+    .. code-block:: python
+
+        from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+        client = BmiClientSingularity(image='docker://ewatercycle/marrmot-grpc4bmi:latest')
+        client.initialize('/opt/MARRMoT/BMI/Config/BMI_testcase_m01_BuffaloRiver_TN_USA.mat')
+        client.update_until(client.get_end_time())
+        del client
+
+    **Example 2: Config file in input directory**
+
+    The config file and all other files the model needs are in a directory (`/tmp/input`).
+    Use `/tmp/work` to capture any output files like logs generated by model.
+
+    .. code-block:: python
+
+        from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+        # Generate config file called 'config.mat' in `/tmp/input` directory
+        client = BmiClientSingularity(image='docker://ewatercycle/marrmot-grpc4bmi:latest',
+                                      input_dirs=['/tmp/input'],
+                                      work_dir='/tmp/work')
+        client.initialize('/tmp/input/config.mat')
+        client.update_until(client.get_end_time())
+        del client
+
+    **Example 3: Read only input directory with config file in work directory**
+
+    The forcing data is in a shared read-only location like `/shared/forcings/walrus`.
+    In the config file (`/tmp/work/walrus.yml`) point to a forcing data file (`/shared/forcings/walrus/PEQ_Hupsel.dat`).
+
+    .. code-block:: python
+
+        from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+        client = BmiClientSingularity(image='docker://ewatercycle/walrus-grpc4bmi:v0.2.0',
+                                      input_dirs=['/shared/forcings/walrus'],
+                                      work_dir='/tmp/work')
+        client.initialize('walrus.yml')
+        client.update_until(client.get_end_time())
+        del client
+
+    **Example 4: Model writes in sub directory of input directory**
+
+    The input directories are mounted read-only so use writable work directory instead.
+    When input directory is read-only to the user then
+    the input dir should be copied to a work directory (`/scratch/wflow`) so model can write.
```

Comment:
This sentence is unclear. Could you please re-phrase the sentence? something like: When the model only works with one directory holding all the input and output sub directories, e.g. wflow model, the user should copy required files to the work directory.